pkg/minio/minio.go

```diff
@@ -0,0 +1,139 @@
+package minio
+
+import (
+	"fmt"
+	"os"
+	"path"
+	"time"
+
+	"k8s.io/apimachinery/pkg/api/resource"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/pkg/api/v1"
+	batchv1 "k8s.io/client-go/pkg/apis/batch/v1"
+)
+
+func waitForCompletedJob(jobName, namespace string, timeout int, cli kubernetes.Interface) error {
+	counter := 0
+	for counter < timeout {
+		j, err := cli.BatchV1().Jobs(namespace).Get(jobName, metav1.GetOptions{})
+		if err != nil {
+			return err
+		}
+		if j.Status.Failed == 1 {
+			err = fmt.Errorf("Unable to run upload job. Received: %s", j.Status.Conditions[0].Message)
+			return err
+		} else if j.Status.Succeeded == 1 {
+			return nil
+		}
+		time.Sleep(time.Duration(time.Second))
+		counter++
+	}
+	return fmt.Errorf("Upload job has not finished after %s seconds", string(timeout))
+}
+
+func validateMaxSize(fileSize int64, cli kubernetes.Interface) error {
+	minioConfig, err := cli.CoreV1().ConfigMaps("kubeless").Get("minio-config", metav1.GetOptions{})
+	if err != nil {
+		return fmt.Errorf("Unable to find Minio configuration. Received: %s", err)
+	}
+	maxFileSize := minioConfig.Data["maxFileSize"]
+	maxSizeQuantity, err := resource.ParseQuantity(maxFileSize)
+	if err != nil {
+		return err
+	}
+	maxSize, _ := maxSizeQuantity.AsInt64()
+	if fileSize > maxSize {
```

Comment:
As mentioned elsewhere, why does this check exist?  If the user wants to use a large file, why are you stopping them?

Reply:
because a malicious user can upload a file of 10 GB file and cause a DoS

Reply:
The only users who can do this are users who already have write access to minio / and-or the RBAC permissions to create Functions that cause writes to minio.  These aren't hostile public users who need to be protected from each other.

Reply:
okay, I was playing the paranoid role here